server/src/main/java/org/apache/kafka/network/metrics/RequestMetrics.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.network.metrics;
+
+import org.apache.kafka.common.network.ClientInformation;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.server.metrics.KafkaMetricsGroup;
+
+import com.yammer.metrics.core.Histogram;
+import com.yammer.metrics.core.Meter;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+
+public class RequestMetrics {
+
+    public static final String CONSUMER_FETCH_METRIC_NAME = ApiKeys.FETCH.name + "Consumer";
+    public static final String FOLLOW_FETCH_METRIC_NAME = ApiKeys.FETCH.name + "Follower";
+    public static final String VERIFY_PARTITIONS_IN_TXN_METRIC_NAME = ApiKeys.ADD_PARTITIONS_TO_TXN.name + "Verification";
+    public static final String REQUESTS_PER_SEC = "RequestsPerSec";
+    public static final String DEPRECATED_REQUESTS_PER_SEC = "DeprecatedRequestsPerSec";
+    public static final String MESSAGE_CONVERSIONS_TIME_MS = "MessageConversionsTimeMs";
+    public static final String TEMPORARY_MEMORY_BYTES = "TemporaryMemoryBytes";
+
+    private static final String REQUEST_QUEUE_TIME_MS = "RequestQueueTimeMs";
+    private static final String LOCAL_TIME_MS = "LocalTimeMs";
+    private static final String REMOTE_TIME_MS = "RemoteTimeMs";
+    private static final String THROTTLE_TIME_MS = "ThrottleTimeMs";
+    private static final String RESPONSE_QUEUE_TIME_MS = "ResponseQueueTimeMs";
+    private static final String RESPONSE_SEND_TIME_MS = "ResponseSendTimeMs";
+    private static final String TOTAL_TIME_MS = "TotalTimeMs";
+    private static final String REQUEST_BYTES = "RequestBytes";
+    private static final String ERRORS_PER_SEC = "ErrorsPerSec";
+
+    // time a request spent in a request queue
+    public final Histogram requestQueueTimeHist;
```

Comment:
Could we add updater for those variables instead of making them public?

Reply:
All of these (apart from `totalTimeHist`) are only accessed by `RequestChannel`. So I thought that this was temporary, and once we migrate `RequestChannel` we will be able to remove the `public` modifier as all callers will be in the same package. Also since these are final fields, having getters doesn't buy us very much.

Ideally we'll also want to refactor `SocketServerTest` so it doesn't use `totalTimeHist`, or move the specific test that uses this field somewhere else.

Reply:
sounds good :)